File: Samples/Android-Advanced/app/src/main/java/com/pushwoosh/sample/DeepLinkActivity.java
package com.pushwoosh.sample;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.Toast;

public class DeepLinkActivity extends Activity {
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		setContentView(R.layout.deep_link);
		setTitle("Deep link activity");
		
		Intent intent = getIntent();
	    String action = intent.getAction();
	    Uri data = intent.getData();
	    
	    if (action == Intent.ACTION_VIEW)
	    {
	    	openUrl(data);
	    }
	}
	
	private void openUrl(Uri uri) {
		String path = uri.getPath();
		Toast.makeText(getApplicationContext(), path, Toast.LENGTH_LONG).show();
	}
}

File: Samples/Android-Advanced-Kotlin/app/src/main/java/com/pushwoosh/sample/DeepLinkActivity.kt
package com.pushwoosh.sample

import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.widget.Toast

class DeepLinkActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.deep_link)
        title = "Deep link activity"

        val intent = intent
        val action = intent.action
        val data = intent.data

        data?.let {
            if (TextUtils.equals(action, Intent.ACTION_VIEW)) {
                openUrl(data)
            }
        } ?: run {
            Log.d(PushwooshSampleApp.LTAG, "Intent data is null")
        }
    }

    private fun openUrl(uri: Uri) {
        val path = uri.path
        Toast.makeText(applicationContext, path, Toast.LENGTH_LONG).show()
    }
}
